JumpscaleCore/tools/threebot/ThreebotExplorer.py

```diff
@@ -161,3 +167,98 @@ def sign(nacl, *args):
+        reservation.data_reservation.expiration_provisioning = int(j.data.time.epoch + 30 * 60)
+        reservation.data_reservation.expiration_reservation = int(j.data.time.epoch + 50 * 60)
+
+        # Create container workload
+        container = reservation.data_reservation.containers.new()
+        container.node_id = "1"
+        container.workload_id = 1
+        container.flist = flist
+        container.hub_url = hub_url
+        container.environment = environment or {}
+        container.entrypoint = entrypoint or ""
+        container.interactive = "yes"  # yes or no
+        reservation.data_reservation.containers.append(container)
+
+        # Create network workload
+        # TODO create network for reservation
+        # network = reservation.data_reservation.networks.new()
+        # network.workload_id = 2
+
+        # stats_aggregator = network.stats_aggregator.new()
+        # stats_aggregator.addr = ""
+        # stats_aggregator.port = 123
+        # stats_aggregator.secret = ""
+        # network.stats_aggregator.append(stats_aggregator)
+
+        # network_resource = network.network_resources.new()
+        # network_resource.node_id = "2"
+        # # hex encoded encrypted with public key of the node, generated by the tfgrid customer
+        # network_resource.wireguard_private_key_encrypted = ""
+        # network_resource.wireguard_public_key = ""
+        # network_resource.wireguard_listen_port = 123
+        # network_resource.iprange = "10.10.10.0/24"
+        # peer = network_resource.peers.new()
+        # peer.public_key = ""
+        # peer.allowed_iprange = ""  #  (Liprange)# is the the same as iprange in the net_resource
+        # peer.endpoint = ""  # optional, only needed to connect out
+        # peer.iprange = "10.10.11.0/24"
+        # network_resource.peers.append(peer)
+        # network.network_resources.append(network_resource)
+
+        # reservation.data_reservation.networks.append(network)
+
+        return reservation
+
+    def container_create(self, flist, hub_url, environment, entrypoint):
+        """
+        Create and register a reservation using explorer workloads actor to deploy a container on a node
+
+        :param flist: link to flist to create a container from
+        :type flist: str
+        :param hub_url: link to hub
+        :type hub_url: str
+        :param environment: environment variables to be passed to the contiainer to be created eg : {"USER":"test_user"}
+        :type environment: dict
+        :param entrypoint: entrypoint to container start
+        :type entrypoint: str
+        :return: reservation object
+        :return type: obj
+        """
+        # Create container reservation
+        reservation = self._reservation_create(flist, hub_url, environment, entrypoint)
+
+        # Create reservation.json
+        reservation.json = j.data.serializers.json.dumps(reservation.data_reservation._ddict)
```

Comment:
I think you can get a json version of the reservation schema with `reservation._json`

Reply:
I didn't find a `_json` attribute for reservation created.